### PR TITLE
Message Grouping

### PR DESCRIPTION
**Pull request by roobscoob**

### CODE REVIEW

File: src/ui/src/channel/message_list.rs [hunk 1]
```diff
@@ -1,50 +1,65 @@
...
     }
 
-    self.real_messages.push(message);
+    let last = self.messages.last_mut();
+
+    if last.is_some() && last.as_ref().unwrap().get_author().get_id() == message.get_author().get_id() {
+      last.unwrap().add(message);
+    } else {
+      self.messages.push(MessageGroup::new(message));
+    }
```
Comment (alii) [inline src/ui/src/channel/message_list.rs]: Could we add a check for if the last message was over 5 minutes ago? Discord will ungroup in that case

**Review by alii — CHANGES_REQUESTED** [review]
Just one thing. rest is great. 🫡

File: src/ui/src/channel/message_list.rs (hunk 1 above)
Comment (roobscoob) [reply src/ui/src/channel/message_list.rs]: aaah, didn't think about that. Will do now

**Review by alii — APPROVED** [review]
lgtm just need to fix the conflict

**Pull request merged**